cpp/src/parquet/arrow/arrow-reader-writer-test.cc

```diff
@@ -2334,19 +2334,20 @@ TEST(TestImpalaConversion, ArrowTimestampToImpalaTimestamp) {
   ASSERT_EQ(expected, calculated);
 }
 
-void TryReadDataFile(const std::string& testing_file_path, bool should_succeed = true) {
-  std::string dir_string(test::get_data_dir());
-  std::stringstream ss;
-  ss << dir_string << "/" << testing_file_path;
-  auto path = ss.str();
-
+void TryReadDataFile(const std::string& path, bool should_succeed = true) {
   auto pool = ::arrow::default_memory_pool();
 
   std::unique_ptr<FileReader> arrow_reader;
   try {
     arrow_reader.reset(new FileReader(pool, ParquetFileReader::OpenFile(path, false)));
     std::shared_ptr<::arrow::Table> table;
-    ASSERT_OK(arrow_reader->ReadTable(&table));
+    auto status = arrow_reader->ReadTable(&table);
+
+    if (should_succeed) {
+      ASSERT_OK(status);
+    } else {
+      ASSERT_FALSE(status.ok()) << "Expected reading file to fail.";
```

Comment:
Perhaps we should test for a specific error using `ASSERT_RAISES`.

Reply:
Thanks for the suggestion, testing for a specific outcome is definitely neater.  I took a different route from using `ASSERT_RAISES`, @pitrou could you take a look at the latest diff and let me know if it looks ok to you?